src/rest.ts

```diff
@@ -1,18 +1,23 @@
 import cockpit from "cockpit";
 
-import { debug } from "./util.js";
+import { debug } from "./util.tsx";
 
-function manage_error(reject, error, content) {
-    let content_o = {};
-    if (content) {
+type JsonObject = cockpit.JsonObject;
+
+// make this `unknown` to conveniently call it on raw error objects
+function format_error(error: object, content: unknown): object {
+    let content_o: JsonObject = {};
+    if (typeof content === 'string') {
         try {
             content_o = JSON.parse(content);
         } catch {
             content_o.message = content;
         }
+        return { ...error, ...content_o };
+    } else {
+        console.warn("format_error(): content is not a string:", content);
+        return error;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2214-c7aa3398-20250801-083853-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)

Reply:
podman API contract.